### ISSUE TITLE
Share a docker-machine on multiples machines

### ISSUE THREAD

**Issue by joliejuke** [docker/machine#3861]
Hi,

I create a amazonec2 docker-machine to deploy an application, so I can manage this docker-machine locally.

But is it possible to share this docker-machine on other machine in order to work with other developers?

Something like a docker-machine pull?

**Comment by easybe** [comment]
I agree, this is one of the biggest flaws of docker-machine. Has nobody come up with a concept for this yet? The solution I found so far to manage a server from multiple clients is to use the same CA certificate by setting `MACHINE_TLS_CA_KEY` and `MACHINE_TLS_CA_CERT` and copying `~/docker/machine/machines/the_machine` between the hosts and editing `config.json` accordingly.

**Comment by hholst80** [comment]
@easybe a first step could be to get `docker-machine` run against a NFS share or similar. Remapping the config directory (`~/.docker/machine`) to some shared place on the network (`/nfs/machine`) do you think that would work?

**Comment by hholst80** [comment]
We could probably make DM configuration files portable if relative paths inside the config where in relation to `MACHINE_CONFIG_DIRECTORY`, which would default to `~/.docker/machine` if not explicitly set to something else.
